letter-sender/src/main/java/com/christmas/letter/sender/service/SnsService.java

```diff
@@ -0,0 +1,53 @@
+package com.christmas.letter.sender.service;
+
+import com.christmas.letter.sender.exceptions.InvalidDataException;
+import com.christmas.letter.sender.model.Letter;
+
+import java.util.regex.Pattern;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+
+@Service
+@RequiredArgsConstructor
+public class SnsService {
+
+  @NonNull
```

Comment:
This is just a preference, but there's no need to keep adding @NonNull wherever we inject something. If such code reaches production, it's clear no one tested even the slightest functionality, so the nonnull part is the smallest concern.